### PR TITLE
refactor: replace lodash clone with object spread

### PR DESCRIPTION
**Pull request by SimonSchick**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Did you update the typescript typings accordingly (if applicable)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

This replaces (almost) all `_.clone` calls with object spreading (there was 1 array clone).

Turns out `_.clone` is around least an order of magnitude slower compared to native object spreading!

Consider this (rather naive) performance test:

```js
'use strict';

const { clone } = require('lodash');

const data = {
  a: 1,
  b: 2,
  c: 3,
  d: 4,
  e: 5
};


let out;
console.time('clone');
for (let i = 0; i < 0xffffff; i++) {
  data[Math.floor(Math.random() * 10)] = Math.random(); //avoid opt
  out = clone(data);
}
console.timeEnd('clone');

console.time('spread');
for (let i = 0; i < 0xffffff; i++) {
  data[Math.floor(Math.random() * 10)] = Math.random(); //avoid opt
  out = { ...data };
}
console.timeEnd('spread');
console.log(out); // avoid opt
```

The results (on my machine):
```
$ node speed.js
clone: 23.944s
spread: 840.681ms
```

See also https://www.measurethat.net/Benchmarks/Show/4542/0/object-cloning-with-lodash-clone-vs-clonedeep-vs-merge

### CODE REVIEW

File: lib/model.js [hunk 1]
```diff
@@ -2598,12 +2593,14 @@ class Model {
       }
       if (options.individualHooks) {
         await Promise.all(instances.map(async instance => {
-          const individualOptions = _.clone(options);
+          const individualOptions = {
+            ...options,
+            validate: false,
+            hooks: true
+          };
           delete individualOptions.fields;
```
Comment (eseliger) [inline lib/model.js:2601]: maybe this can leverage a `omit` on the options, bc delete is slow

File: lib/model.js [hunk 2]
```diff
@@ -3140,10 +3137,12 @@ class Model {
           }
         } else {
           instances = await Promise.all(instances.map(async instance => {
-            const individualOptions = _.clone(options);
+            const individualOptions = {
+              ...options,
+              hooks: false,
+              validate: false
+            };
             delete individualOptions.individualHooks;
```
Comment (eseliger) [inline lib/model.js:3145]: same here

File: lib/model.js (hunk 1 above)
Comment (SimonSchick) [reply lib/model.js:2601]: `omit` creates a new object entirely which is MUCH slower.

Comment (SimonSchick) [reply lib/model.js:2601]: I'd agree if this was more than 2-3 properties but for a single one cloning the entire object just isn't worth it.

**Comment by SimonSchick** [comment]
More deadlocking 🎉

**Pull request merged**